internal/provider/resource_subaccount_environment_instance.go

```diff
@@ -260,20 +275,46 @@ func (rs *subaccountEnvironmentInstanceResource) Update(ctx context.Context, req
 		return
 	}
 
-	resp.Diagnostics.AddError("API Error Updating Resource Environment Instance (Subaccount)", "Update is not yet implemented.")
-
-	/* TODO: implementation of UPDATE operation
-	cliRes, err := gen.client.Execute(ctx, btpcli.Update, gen.command, plan)
+	_, _, err := rs.cli.Accounts.EnvironmentInstance.Update(ctx,
+		plan.SubaccountId.ValueString(),
+		plan.Id.ValueString(),
+		plan.PlanName.ValueString(),
+		plan.Parameters.ValueString(),
```

Comment:
To stay consistent with the CREATE scenario, would it make sense to embed the parameters in a struc (which would then also need to be adjusted in the facade)

Reply:
I was not able to observe a pattern when a struct is used. My impression is that it depends on the number of parameters. In this case, the create needs a lot more parameters and I was fine with having four of them in the signature of update. But if you prefer a struct, then I can change this.

Reply:
Indeed we are not consistent over the resources. However I would stay consistent within one resource and pass the parameters via a struc

Reply:
Done